src/character.h

```diff
@@ -484,6 +484,7 @@ class Character : public Creature, public visitable<Character>
         std::vector<bionic> my_bionics;
 
     protected:
+        virtual void on_stat_change( const std::string &, int ) override {};
```

Comment:
The reason there are so many string IDs is for ease of use with strings
loaded from json. Anything that needs an ID and is entirely defined within
the code should use an enum.